File: app/src/main/java/be/vergauwen/simon/cleanmvp/di/ApplicationModule.java
package be.vergauwen.simon.cleanmvp.di;

import android.content.Context;
import be.vergauwen.simon.cleanmvp.MVPApplication;
import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {
    private MVPApplication application;

    public ApplicationModule(MVPApplication application) {
        this.application = application;
    }

    @Provides
    @ApplicationScope
    public Context provideApplicationContext() {
        return application;
    }
}

File: app/src/main/kotlin/be/vergauwen/simon/di/ApplicationModule.kt
package be.vergauwen.simon.di

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(private val application: Application) {

  @ApplicationScope
  @Provides
  fun provideApplicationContext(): Context {
    return application
  }
}